server/index.js

```diff
@@ -62,6 +62,106 @@ app.get('/api/products/:productId', (req, res, next) => {
 }
 );
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    return res.status(200).json([]);
+  } else {
+    const sqlSelect = `
+  select "c"."cartItemId",
+        "c"."price",
+        "p"."productId",
+        "p"."image",
+        "p"."name",
+        "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+    where "c"."cartId" = $1
+    `;
+    const queryValue = [req.session.cartId];
+    db.query(sqlSelect, queryValue)
+      .then(result => {
+        res.json(result.rows);
+      })
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  if (isNaN(req.body.productId)) {
+    return res.status(400).json({ error: `Invalid field used for this POST method for ProductId '${req.body.productId}'. Please correct property syntax or try using a number type value.` });
+  }
+  if (req.body.productId < 0) {
+    return res.status(400).json({ error: `Invalid field used for this POST method for ProductId '${req.body.productId}'. Please use an Id greater than 0.` });
+  }
+  const sqlSelect = `
+  select
+  "price",
+  "productId"
+  from
+    "products"
+  where "productId" = $1
+  `;
+  const values = [req.body.productId];
+  db.query(sqlSelect, values)
+    .then(result => {
+      if (result.rows.length === 0) {
+        throw next(new ClientError(`Product Id: ${req.body.productId} cannot be found`, 400));
```

Comment:
Pretty close. `throw` is definitely the right thing to do here, but since your `catch()` already sends the error to `next()`, we don't want to double-call `next()` by default. You can remove the call to `next()` here.

```js
        throw new ClientError(`Product Id: ${req.body.productId} cannot be found`, 400);

```